js/sign/src/utils/utils.ts

```diff
@@ -31,15 +36,63 @@ export function parsePemKey(
   });
 }
 
-export function getRawPublicKey(publicKey: crypto.KeyObject) {
-  // Currently this is the only way for us to get the raw 32 bytes of the public key.
-  return new Uint8Array(
-    publicKey.export({ type: 'spki', format: 'der' }).slice(-32)
+export function isAsymmetricKeyTypeSupported(key: crypto.KeyObject) {
+  return (
+    key.asymmetricKeyType === 'ed25519' ||
+    (key.asymmetricKeyType === 'ec' &&
+      key.asymmetricKeyDetails!.namedCurve === 'secp256k1')
+  );
+}
+
+export enum SignatureType {
+  Ed25519,
+  EcdsaP256SHA256,
+}
+
+export function getSignatureType(key: crypto.KeyObject) {
+  assert(
+    isAsymmetricKeyTypeSupported(key),
+    'Expected either "Ed25519" or "ECDSA P-256" key.'
   );
+  if (key.asymmetricKeyType === 'ed25519') {
+    return SignatureType.Ed25519;
+  }
+  return SignatureType.EcdsaP256SHA256;
```

Comment:
Using a `switch` with a `default` case would feel a bit safer here, IMHO.

Reply:
A switch on what exactly? 🤔 

- `SignatureType.Ed25519 <=> key.asymmetricKeyType === 'ed25519'`
- `SignatureType.EcdsaP256SHA256 <=> key.asymmetricKeyType === 'ec' &&
      key.asymmetricKeyDetails?.namedCurve === 'secp256k1'`

Reply:
I would think @cmfcmf means something like this

`
switch 
`

Reply:
I thought of something like

```js
switch (key.asymmetricKeyType) {
  case 'ed25519':
    return SignatureType.Ed25519;
  case 'ec':
    if (key.asymmetricKeyDetails?.namedCurve === 'secp256k1') {
      return SignatureType.EcdsaP256SHA256;
    }
    assert(false);
  default:
    assert(false);
```

I think ideally you'd refactor the functions to this:

```
function maybeGetSignatureType(key: crypto.KeyObject): SignatureType | null;
export function getSignatureType(key: crypto.KeyObject): SignatureType {
  const s = maybeGetSignatureType(key);
  assert(s);
  return s;
}
export function isAsymmetricKeyTypeSupported(key: crypto.KeyObject): bool {
  return maybeGetSignatureType(key) !== null;
}
```

Reply:
Done!